packages/iov-bns/src/bnsconnection.ts

```diff
@@ -332,14 +332,14 @@ export class BnsConnection implements AtomicSwapConnection {
   public async getSwapsFromState(query: AtomicSwapQuery): Promise<ReadonlyArray<AtomicSwap>> {
     const doQuery = async (): Promise<QueryResponse> => {
       if (isAtomicSwapIdQuery(query)) {
-        return this.query("/escrows", query.swapid.data);
+        return this.query("/escrows", query.id.data);
       } else if (isAtomicSwapSenderQuery(query)) {
         return this.query("/escrows/sender", decodeBnsAddress(query.sender).data);
       } else if (isAtomicSwapRecipientQuery(query)) {
         return this.query("/escrows/recipient", decodeBnsAddress(query.recipient).data);
       } else {
-        // if (isQueryBySwapHash(query))
-        return this.query("/escrows/arbiter", hashIdentifier(query.hashlock));
+        // if (isAtomicSwapHashQuery(query))
```

Comment:
Why not make this logic explicit?

Reply:
The type system ensures that in the `else` block, `query` is of type `AtomicSwapHashQuery`. If we eyplicitly add `else if (isAtomicSwapHashQuery(query))`, we create a new dead code branch with `query` of type `never`. Not sure if that would be nicer than the current code.

Reply:
I think it would be easy to read and also guards against the situation where we develop the type further and don't update this code (we can give a nice error message rather than something about the `hash` property being missing).

Reply:
ok, https://github.com/iov-one/iov-core/pull/947/commits/b80d111b38395a836ef2b255d0472886bf8d508d